app/controllers/api/connect/v3/systems/products_controller.rb

```diff
@@ -48,18 +48,15 @@ def offline_migrations
   end
 
   def upgrade
-    activation = @system.activations.joins(:service).find_by('services.product_id': [@product.id, @product.predecessor_ids, @product.successor_ids].flatten)
+    obsoleted_product_ids = ([@product] + @product.predecessors + @product.successors).map(&:id)
+    obsoleted_service = @system.services.find_by(product_id: obsoleted_product_ids)
+    @obsolted_service_name = obsoleted_service.name if obsoleted_service
```

Comment:
typo in `@obsoleted_...`

Reply:
How are tests passing with such a typo? I fear we're not testing something

Reply:
That doesn't seem to have been tested ever but was coded in `render_service`, I will see if I can throw one in.